### PR TITLE
fetchzip, fetchgit: cleanup handling of optional features and whitespace

### PR DESCRIPTION
**Pull request by bhipple**
No intended functional difference, just trying to polish to make this easier to
extend and more aligned with other nix conventions.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/fetchzip/default.nix [hunk 1]
```diff
@@ -5,47 +5,46 @@
...
   recursiveHash = true;
-
   downloadToTemp = true;
 
-  postFetch =
-    ''
-      unpackDir="$TMPDIR/unpack"
-      mkdir "$unpackDir"
-      cd "$unpackDir"
+  postFetch = ''
```
Comment (bhipple) [inline pkgs/build-support/fetchzip/default.nix]: This should do the same thing without any hash changes or rebuilds, as verified by doing `nix-build -A <something>.src --check` for a bunch of <something>s that use various fetchers.  Same for the fetchgit whitespace indentation cleanup.

**Comment by bhipple** [comment]
@GrahamcOfBorg eval

**Comment by aanderse** [comment]
(triage) ping

**Comment by bhipple** [comment]
Clearly did not test well enough; I reverted the whole PR for the meantime, apologies.

**Comment by aanderse** [comment]
Don't worry, it happens from time to time! 😃

**Pull request merged**